Description: Skip failing bindgen tests on 32-bit arches 
Author: Matthias Geiger <werdahias@riseup.net>
Forwarded: https://github.com/NoXF/libwebp-sys/issues/22 
Last-Update: 2023-09-10
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/

--- a/src/ffi.rs
+++ b/src/ffi.rs
@@ -708,6 +708,7 @@ pub struct WebPMemoryWriter {
     pub pad: [u32; 1usize],
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPMemoryWriter() {
     const UNINIT: ::core::mem::MaybeUninit<WebPMemoryWriter> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -842,6 +843,7 @@ pub struct WebPPicture {
     pub pad7: [*mut ::core::ffi::c_void; 2usize],
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPPicture() {
     const UNINIT: ::core::mem::MaybeUninit<WebPPicture> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -1450,6 +1452,7 @@ pub struct WebPRGBABuffer {
     pub size: usize,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPRGBABuffer() {
     const UNINIT: ::core::mem::MaybeUninit<WebPRGBABuffer> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -1511,6 +1514,7 @@ pub struct WebPYUVABuffer {
     pub a_size: usize,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPYUVABuffer() {
     const UNINIT: ::core::mem::MaybeUninit<WebPYUVABuffer> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -1663,6 +1667,7 @@ pub union WebPDecBuffer__bindgen_ty_1 {
     pub YUVA: WebPYUVABuffer,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPDecBuffer__bindgen_ty_1() {
     const UNINIT: ::core::mem::MaybeUninit<WebPDecBuffer__bindgen_ty_1> =
         ::core::mem::MaybeUninit::uninit();
@@ -1704,6 +1709,7 @@ impl ::core::fmt::Debug for WebPDecBuffe
     }
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPDecBuffer() {
     const UNINIT: ::core::mem::MaybeUninit<WebPDecBuffer> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -2183,6 +2189,7 @@ pub struct WebPDecoderConfig {
     pub options: WebPDecoderOptions,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPDecoderConfig() {
     const UNINIT: ::core::mem::MaybeUninit<WebPDecoderConfig> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -2285,6 +2292,7 @@ pub struct WebPData {
     pub size: usize,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPData() {
     const UNINIT: ::core::mem::MaybeUninit<WebPData> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -2399,6 +2407,7 @@ pub struct WebPMuxFrameInfo {
     pub pad: [u32; 1usize],
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPMuxFrameInfo() {
     const UNINIT: ::core::mem::MaybeUninit<WebPMuxFrameInfo> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -2792,6 +2801,7 @@ pub struct WebPIterator {
     pub private_: *mut ::core::ffi::c_void,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPIterator() {
     const UNINIT: ::core::mem::MaybeUninit<WebPIterator> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
@@ -2972,6 +2982,7 @@ pub struct WebPChunkIterator {
     pub private_: *mut ::core::ffi::c_void,
 }
 #[test]
+#[cfg(not(target_pointer_width = "32"))]
 fn bindgen_test_layout_WebPChunkIterator() {
     const UNINIT: ::core::mem::MaybeUninit<WebPChunkIterator> = ::core::mem::MaybeUninit::uninit();
     let ptr = UNINIT.as_ptr();
